### PR TITLE
Convert demo application to Vite

### PR DESCRIPTION
**Pull request by jonkoops**
Replaces WebPack with Vite to build the demo app, this reduces the amount of dependencies and is also a faster and more consistent development experience. Works towards closing #7287

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -136,6 +136,14 @@
                 "react/jsx-key": "off",
                 "no-console": "off"
             }
+        },
+        {
+            "files": ["packages/react-integration/demo-app-ts/**/*"],
+            "rules": {
+                "patternfly-react/no-anonymous-functions": "off",
```
Comment (jonkoops) [inline .eslintrc.json:143]: This linting rule gives false positives for class components that already have a displayName, so I've turned this off. For example:

```ts
class MyComponent extends React.Component {
  static displayName = "MyComponent"; // It doesn't like this.
}
```

File: .eslintrc.json [hunk 2]
```diff
@@ -136,6 +136,14 @@
...
+                "react/react-in-jsx-scope": "off",
```
Comment (jonkoops) [inline .eslintrc.json:144]: There is no need for this, as Vite uses the [JSX transform](https://legacy.reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html).

File: .eslintrc.json [hunk 3]
```diff
@@ -136,6 +136,14 @@
...
+                "spaced-comment": "off"
```
Comment (jonkoops) [inline .eslintrc.json:145]: This gives a false positive, so I've disabled it for now.

File: package.json [hunk 4]
```diff
@@ -93,11 +93,10 @@
     "lint:tests": "yarn lint packages/*/src/components/*/__tests__/*.test.*",
     "prepare": "ts-patch install -s",
     "serve:docs": "yarn workspace @patternfly/react-docs serve",
-    "serve:integration": "lerna run serve:demo-app",
+    "serve:integration": "lerna run preview --scope=demo-app-ts",
     "start": "yarn build && concurrently --kill-others \"yarn watch\" \"yarn workspace @patternfly/react-docs develop\"",
     "start:cypress": "lerna run cypress:open",
-    "start:demo-app": "lerna run start:demo-app --stream",
-    "start:demo-app:hot": "lerna run start:demo-app:hot --stream",
```
Comment (jonkoops) [inline package.json:100]: Hot reloading is default in Vite during development, so it is no longer needed to keep this separate script around.

File: packages/react-integration/demo-app-ts/src/App.test.tsx
Comment (jonkoops) [inline packages/react-integration/demo-app-ts/src/App.test.tsx:1]: This test never ran, so I removed it.

**Review by nicolethoen — APPROVED** [review]

**Review by tlabaj — APPROVED** [review]
LGTM

**Review by kmcfaul — APPROVED** [review]

**Pull request merged**